### ISSUE TITLE
Desync can cause users to hear others when they have their whisper bubble enabled

### ISSUE THREAD

**Issue by epicEaston197** [Yellow-Dog-Man/Resonite-Issues#1611]
### Describe the bug?

If someone else turns on whisper mode and another user is still synced they will still be able to hear the user that has their bubble on

### To Reproduce

Go into a populated world (preferably australia) there will always be a user that will be desynced and we'll be able to hear others when they have their bubbles enabled

### Expected behavior

whisper bubble should have some network priority so it maintains functionality

### Resonite Version Number

2024.4.3.1170

### What Platforms does this occur on?

Windows

### What headset if any do you use?

Desktop

### Log Files

It didn't happen to me it happened to another user so I just thought I would note it down here

### Reporters

@epicEaston197

**Comment by ProbablePrime** [comment]
As a reminder, whisper mode is not a secure communication channel and should be used literally. You want to whisper something to someone in a public space? That's what it is for, not secure communications.

**Comment by epicEaston197** [comment]
> queued packets

Yes queued packets

**Comment by epicEaston197** [comment]
> As a reminder, whisper mode is not a secure communication channel and should be used literally. You want to whisper something to someone in a public space? That's what it is for, not secure communications.

I am aware that's why I said nothing about secure communication in the original issue I just wanted to note it down as
a "this is a thing that can happen and people should probably be aware of it"

**Comment by Frooxius** [comment]
There's a few things:
- Like Prime said, whisper is not designed as privacy feature, but more accessibility
- We do want to focus on resolving queued packets where possible
- This is a bit tricky issue, but there are some things that are planned to do - selectively sending stream data to users. This would make the user indicate who gets their voice data based on their own state, so if they see someone as outside the whisper bubble, they will not send voice data in them. This would improve these kinds of degraded situations, while also improving network performance

**Comment by Frooxius** [comment]
Yes it'll make it harder to eavesdrop. You'd only be able to do it if the host client was modified to send the data to everyone or if the host wanted to eavesdrop.

**Comment by Stellanora64** [comment]
This also happens when a user is muted, as it works by disabling the component which then becomes a queued packet, thus when someone is queueing packets they here a person when they are ment to be mute, and then later that user becomes muted when they are no longer muted.

**Comment by epicEaston197** [comment]
> This also happens when a user is muted, as it works by disabling the component which then becomes a queued packet, thus when someone is queueing packets they hear a person when they are ment to be mute, and then later that user becomes muted when they are no longer muted.
> 
> This issue, along with other issues related to queued packets, was already discussed here: #120

I think this is fit for separate issue

**Comment by JackTheFoxOtter** [comment]
> I think this is fit for separate issue

I mean, it's basically the same underlying issue, isn't it?

**Comment by Frooxius** [comment]
@Stellanora64 Were you able to actually replicate that?

When user mutes themselves on the dash, we actually stop their audio at the driver level completely. This makes it that it doesn't matter if they're queued or not - there's no audio data at all for the period they're muted.

If you are still hearing users when they have muted themselves, this would indicate there is something wrong with the mechanism, which should be a separate report.

**Comment by Stellanora64** [comment]
@Frooxius Yes, or at least of what I remember. I don't have any logs or videos reproducing it so I won't be able to create an issue for it yet. I will try and reproduce it and create an issue for it during the CJ tomorrow as I'm consistently desynced there.
